From 480ce066abe889e633dbc5c011083503ceea1f95 Mon Sep 17 00:00:00 2001
From: florent pirou <florent.pirou@intel.com>
Date: Tue, 28 Apr 2026 05:50:59 -0700
Subject: [PATCH 58/62] ipu6-isys : add d4xx pixel format support

* add missing D4XX calibration and metadata format
* ordered Z16 before UYVY

Signed-off-by: florent pirou <florent.pirou@intel.com>
---
 ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-fw-isys.h   | 1 +
 ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-csi2.c | 2 ++
 .../drivers/media/pci/intel/ipu6/ipu6-isys-subdev.c        | 2 ++
 .../drivers/media/pci/intel/ipu6/ipu6-isys-video.c         | 7 +++++--
 4 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-fw-isys.h b/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-fw-isys.h
index 545180c..0ec08ee 100644
--- a/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-fw-isys.h
+++ b/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-fw-isys.h
@@ -209,6 +209,7 @@ enum ipu6_fw_isys_frame_format_type {
 	IPU6_FW_ISYS_FRAME_FORMAT_RGBA888, /* 32 bit RGBA, 1 plane, A=Alpha */
 	IPU6_FW_ISYS_FRAME_FORMAT_QPLANE6, /* Internal, for advanced ISP */
 	IPU6_FW_ISYS_FRAME_FORMAT_BINARY_8, /* byte stream, used for jpeg. */
+	IPU6_FW_ISYS_FRAME_FORMAT_VYUY,	/* 16 bit YUV 422, VYUY interleaved */
 	N_IPU6_FW_ISYS_FRAME_FORMAT
 };
 
diff --git a/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-csi2.c b/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-csi2.c
index 89ae50e..a1180bf 100644
--- a/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-csi2.c
+++ b/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-csi2.c
@@ -44,11 +44,13 @@ static const u32 csi2_supported_codes[] = {
 	MEDIA_BUS_FMT_SRGGB8_1X8,
 	MEDIA_BUS_FMT_VYUY8_1X16, /* D4XX specific */
 	MEDIA_BUS_FMT_Y8_1X8, /* D4XX specific */
+	MEDIA_BUS_FMT_Y10_1X10, /* D4XX specific */
 	MEDIA_BUS_FMT_META_8,
 	MEDIA_BUS_FMT_META_10,
 	MEDIA_BUS_FMT_META_12,
 	MEDIA_BUS_FMT_META_16,
 	MEDIA_BUS_FMT_META_24,
+	MEDIA_BUS_FMT_FIXED,
 	0
 };
 
diff --git a/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-subdev.c b/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-subdev.c
index 1911466..4bdcfa5 100644
--- a/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-subdev.c
+++ b/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-subdev.c
@@ -34,6 +34,7 @@ unsigned int ipu6_isys_mbus_code_to_bpp(u32 code)
 	case MEDIA_BUS_FMT_SRGGB12_1X12:
 	case MEDIA_BUS_FMT_META_12:
 		return 12;
+	case MEDIA_BUS_FMT_Y10_1X10:
 	case MEDIA_BUS_FMT_SBGGR10_1X10:
 	case MEDIA_BUS_FMT_SGBRG10_1X10:
 	case MEDIA_BUS_FMT_SGRBG10_1X10:
@@ -74,6 +75,7 @@ unsigned int ipu6_isys_mbus_code_to_mipi(u32 code)
 	case MEDIA_BUS_FMT_SGRBG12_1X12:
 	case MEDIA_BUS_FMT_SRGGB12_1X12:
 		return MIPI_CSI2_DT_RAW12;
+	case MEDIA_BUS_FMT_Y10_1X10:
 	case MEDIA_BUS_FMT_SBGGR10_1X10:
 	case MEDIA_BUS_FMT_SGBRG10_1X10:
 	case MEDIA_BUS_FMT_SGRBG10_1X10:
diff --git a/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-video.c b/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-video.c
index b3d1a39..5a60167 100644
--- a/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-video.c
+++ b/ipu6-drivers/drivers/media/pci/intel/ipu6/ipu6-isys-video.c
@@ -80,6 +80,8 @@ const struct ipu6_isys_pixelformat ipu6_isys_pfmts[] = {
 	  IPU6_FW_ISYS_FRAME_FORMAT_RAW10 },
 	{ V4L2_PIX_FMT_SRGGB10P, 10, 10, MEDIA_BUS_FMT_SRGGB10_1X10,
 	  IPU6_FW_ISYS_FRAME_FORMAT_RAW10 },
+	{ V4L2_PIX_FMT_Z16, 16, 16, MEDIA_BUS_FMT_UYVY8_1X16,
+	  IPU6_FW_ISYS_FRAME_FORMAT_UYVY }, /* D4XX specific */
 	{ V4L2_PIX_FMT_UYVY, 16, 16, MEDIA_BUS_FMT_UYVY8_1X16,
 	  IPU6_FW_ISYS_FRAME_FORMAT_UYVY},
 	{ V4L2_PIX_FMT_YUYV, 16, 16, MEDIA_BUS_FMT_YUYV8_1X16,
@@ -89,14 +91,14 @@ const struct ipu6_isys_pixelformat ipu6_isys_pfmts[] = {
 	{ V4L2_PIX_FMT_BGR24, 24, 24, MEDIA_BUS_FMT_RGB888_1X24,
 	  IPU6_FW_ISYS_FRAME_FORMAT_RGBA888 },
 	/* D4XX specific */
-	{ V4L2_PIX_FMT_Z16, 16, 16, MEDIA_BUS_FMT_UYVY8_1X16,
-	  IPU6_FW_ISYS_FRAME_FORMAT_UYVY },
 	{ V4L2_PIX_FMT_Y8I, 16, 16, MEDIA_BUS_FMT_VYUY8_1X16,
 	  IPU6_FW_ISYS_FRAME_FORMAT_UYVY },
 	{ V4L2_PIX_FMT_Y12I, 32, 24, MEDIA_BUS_FMT_RGB888_1X24, /* TODO: To be verified */
 	  IPU6_FW_ISYS_FRAME_FORMAT_RGBA888 },
 	{ V4L2_PIX_FMT_GREY, 8, 8, MEDIA_BUS_FMT_Y8_1X8,
 	  IPU6_FW_ISYS_FRAME_FORMAT_RAW8 },
+	{ V4L2_PIX_FMT_Y10, 16, 10, MEDIA_BUS_FMT_Y10_1X10,
+	  IPU6_FW_ISYS_FRAME_FORMAT_RAW16 },
 	{ V4L2_META_FMT_GENERIC_8, 8, 8, MEDIA_BUS_FMT_META_8,
 	  IPU6_FW_ISYS_FRAME_FORMAT_RAW8, true },
 	{ V4L2_META_FMT_GENERIC_CSI2_10, 10, 10, MEDIA_BUS_FMT_META_10,
@@ -105,6 +107,7 @@ const struct ipu6_isys_pixelformat ipu6_isys_pfmts[] = {
 	  IPU6_FW_ISYS_FRAME_FORMAT_RAW12, true },
 	{ V4L2_META_FMT_GENERIC_CSI2_16, 16, 16, MEDIA_BUS_FMT_META_16,
 	  IPU6_FW_ISYS_FRAME_FORMAT_RAW16, true },
+	{ V4L2_META_FMT_D4XX, 8, 8, MEDIA_BUS_FMT_FIXED, 0 },
 };
 
 static int video_open(struct file *file)
-- 
2.43.0

